### DISCUSSION TITLE
Set `python.pythonPath` dynamically

### DISCUSSION THREAD

**Discussion by kevinli1993** [yegappan/lsp#541] [Q&A]
Hi! The wiki gives the following example for setting up `pyright`:
```
call LspAddServer([#{name: 'pyright',
                 \   filetype: 'python',
                 \   path: '/usr/local/node_modules/.bin/pyright-langserver',
                 \   args: ['--stdio'],
                 \   workspaceConfig: #{
                 \     python: #{
                 \       pythonPath: '/usr/bin/python3.10'
                 \   }}
                 \ }])
```

What I would like to achieve is to set `python.pythonPath` dynamically, e.g. based on the path of the buffer being visited. 

In general, is it possible to set the options for LspAddServer dynamically/programmatically, perhaps based on autocmd's?

**Comment by ddominguez** [comment]
I've been using ```workspaceConfig: #{python: #{pythonPath: exepath('python')}}```

It will use my virtual environment's python path if it is activated.
